### DISCUSSION TITLE
No error reported on auto-merge failure due to HTTP 405 (Method Not Allowed)

### DISCUSSION THREAD

**Discussion by rpdelaney** [renovatebot/renovate#23255] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

There are two problems here:

1. Auto-merge fails with HTTP 405 (Method Not Allowed). [Job log](https://developer.mend.io/github/trussworks/template-tech-challenge/-/job/018930de-338f-7c02-873d-a21484f16cf2)
2. No error is reported on the pull request.

In my opinion, the significance of the second problem should not be underestimated. The lack of error reporting made this very mysterious and difficult to troubleshoot. It would be much better to report that an error occurred in a comment with a link to the developer logs.

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Failed to merge commit PR (branch="renovate/docker-setup-buildx-action-2.x")
{
  "err": {
    "name": "HTTPError",
    "code": "ERR_NON_2XX_3XX_RESPONSE",
    "timings": {
      "start": 1688750702050,
      "socket": 1688750702050,
      "lookup": 1688750702050,
      "connect": 1688750702053,
      "secureConnect": 1688750702058,
      "upload": 1688750702058,
      "response": 1688750702550,
      "end": 1688750702550,
      "phases": {
        "wait": 0,
        "dns": 0,
        "tcp": 3,
        "tls": 5,
        "request": 0,
        "firstByte": 492,
        "download": 0,
        "total": 500
      }
    },
    "message": "Response code 405 (Method Not Allowed)",
    "stack": "HTTPError: Response code 405 (Method Not Allowed)\n    at Request.<anonymous> (/opt/containerbase/tools/renovate/35.159.7/node_modules/got/dist/source/as-promise/index.js:118:42)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)",
    "options": {
      "headers": {
        "user-agent": "RenovateBot/35.159.7 (https://github.com/renovatebot/renovate)",
        "accept": "application/json, application/vnd.github.machine-man-preview+json",
        "authorization": "***********",
        "content-type": "application/json",
        "content-length": "24",
        "accept-encoding": "gzip, deflate, br"
      },
      "url": "https://api.github.com/repos/trussworks/template-tech-challenge/pulls/228/merge",
      "hostType": "github",
      "username": "",
      "password": "",
      "method": "PUT",
      "http2": false
    },
    "response": {
      "statusCode": 405,
      "statusMessage": "Method Not Allowed",
      "body": {
        "message": "You're not authorized to push to this branch. Visit https://docs.github.com/articles/about-protected-branches/ for more information.",
        "documentation_url": "https://docs.github.com/articles/about-protected-branches"
      },
      "headers": {
        "server": "GitHub.com",
        "date": "Fri, 07 Jul 2023 17:25:02 GMT",
        "content-type": "application/json; charset=utf-8",
        "content-length": "226",
        "x-github-media-type": "github.v3; param=machine-man-preview",
        "x-github-api-version-selected": "2022-11-28",
        "x-ratelimit-limit": "12500",
        "x-ratelimit-remaining": "12473",
        "x-ratelimit-reset": "1688753425",
        "x-ratelimit-used": "27",
        "x-ratelimit-resource": "core",
        "access-control-expose-headers": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset",
        "access-control-allow-origin": "*",
        "strict-transport-security": "max-age=31536000; includeSubdomains; preload",
        "x-frame-options": "deny",
        "x-content-type-options": "nosniff",
        "x-xss-protection": "0",
        "referrer-policy": "origin-when-cross-origin, strict-origin-when-cross-origin",
        "content-security-policy": "default-src 'none'",
        "vary": "Accept-Encoding, Accept, X-Requested-With",
        "x-github-request-id": "79A5:339F:38E1D0D:737B7DF:64A84A6E",
        "connection": "close"
      },
      "httpVersion": "1.1",
      "retryCount": 0
    }
  }
}

INFO: All merge attempts failed (branch="renovate/docker-setup-buildx-action-2.x")
{
  "pr": 228
}
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
PRs welcome if you can find a way to add some info in a PR comment

**Reply by rpdelaney** [reply]
I'll dig around and see if I can come up with something. Since we're not self-hosting it won't be easy to test, though.

In the mean time, if anyone can make suggestions for how to get these auto-merging again we'd sure appreciate it! 😅

**Reply by rarkins** [reply]
`message": "You're not authorized to push to this branch. Visit https://docs.github.com/articles/about-protected-branches/ for more information.",`

**Reply by rpdelaney** [reply]
I saw that. What's not completely clear is why renovatebot isn't authorized. Does it require a specific merge strategy to be allowed? We don't use squash or rebase:

![Screenshot 2023-07-07 at 16 07 31](https://github.com/renovatebot/renovate/assets/1139517/4f2a4c8d-08a0-4655-aee7-f3ccb83885dc)

**Reply by rarkins** [reply]
Please screenshot the full branch protection page

**Reply by rpdelaney** [reply]
<details><summary>big screenshot</summary>

![FireShotCapture-github com-011_Settings · Branches · Branch protection rule · trussworks_template-tech__github com](https://github.com/renovatebot/renovate/assets/1139517/c50ec11e-0c74-4761-a4c3-ff0a124388eb)

</details>

**Reply by rarkins** [reply, marked as answer]
It's the "restrict who can push" rule.

**Reply by rpdelaney** [reply]
![Screenshot 2023-07-10 at 12 54 27](https://github.com/renovatebot/renovate/assets/1139517/6e4c466b-e6a9-4226-8ddb-964e62a826e2)

For anyone who finds this from a search, renovatebot has to be granted push access to `main`.
